compiler/rustc_error_messages/src/lib.rs

```diff
@@ -111,6 +111,13 @@ pub fn fluent_bundle(
     trace!(?locale);
     let mut bundle = FluentBundle::new(vec![locale]);
 
+    // Fluent diagnostics can insert directionality isolation markers around interpolated variables
+    // indicating that there may be a shift from right-to-left to left-to-right text (or
+    // vice-versa). These are disabled because they are sometimes visible in the error output, but
+    // may be worth investigating in future (for example: if type names are left-to-right and the
+    // surrounding diagnostic messages are right-to-left, then these might be helpful).
+    bundle.set_use_isolating(false);
```

Comment:
thought: yeah the issue here is that many terminals don't render this, but e.g. things like VSCode will. Might need to be a configured option.

Reply:
I've added a `-Z` flag to control this for now.